api/hypershift/v1beta1/aws.go

```diff
@@ -59,8 +59,21 @@ type AWSNodePoolPlatform struct {
 	Placement *PlacementOptions `json:"placement,omitempty"`
 }
 
+// AWSSpotMarketOptions defines configuration for AWS Spot instances
+type AWSSpotMarketOptions struct {
+	// maxPrice defines the maximum price (USD per instance‑hour) you are willing to pay for a Spot instance.
+	// If omitted, the On‑Demand price is used as the ceiling.
```

Comment:
>// If omitted, the On‑Demand price is used as the ceiling.

How can I express intent for this with this API?

Reply:
"If you specify a maximum price, it must be more than USD $0.001. Specifying a value below USD $0.001 will result in an InvalidParameterValue error message."
https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_SpotMarketOptions.html

Reply:
Your pattern presently allows `0` as a valid choice, please update to remove that